ctc_lib/logging/log_manager.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/GoogleCloudPlatform/runtimes-common/ctc_lib/constants"
+	rotatelogs "github.com/lestrrat-go/file-rotatelogs"
+	log "github.com/sirupsen/logrus"
+)
+
+func NewLogger(dir string, prefix string, level log.Level, enableColors bool) *log.Logger {
+	if level == log.DebugLevel {
+		// Log to File when verbosity=debug
+		logging_dir, err := ioutil.TempDir(dir, prefix)
+		if err != nil {
+			return handleFileLoggingError(err, level, enableColors)
+		}
+		path := filepath.Join(logging_dir, constants.LogFileName)
+		writer, err := rotatelogs.New(
+			path+".%Y%m%d%H%M",
+			rotatelogs.WithLinkName(path),
+			rotatelogs.WithMaxAge(time.Duration(86400)*time.Second),
```

Comment:
also, maybe we should make these configurable? either through a flag or through a config file.